### PR TITLE
Adding auth join workflow for LinkedIn and Google

### PR DESCRIPTION
**Pull request by eparubrub**
- adds LinkedIn and Google authentication workflow for the join workflow
- auto-fills the email and name fields for both LinkedIn and Google
- gives the user an option to save their LinkedIn profile picture data

**DO NOT MERGE UNTIL [LINKEDIN OAUTH REDIRECT URL](https://learn.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow?tabs=HTTPS1#step-1-configure-your-application) HAS BEEN SET** -

### CODE REVIEW

File: pages/join/01-you.tsx [hunk 1]
```diff
@@ -18,13 +23,57 @@ export async function getStaticProps() {
...
+    sessionStorage.setItem(SessionStorageEnum.PREVIOUS_PAGE, "/join/01-you");
+    router.push({ pathname: `/login` });
+  } else {
+    setName &&
+      setName(
+        capitalizeFirstLetters(
+          sessionStorage.getItem(SessionStorageEnum.USER_NAME)
+        )
+      );
+    setEmail && setEmail(sessionStorage.getItem(SessionStorageEnum.USER_EMAIL));
```
Comment (eparubrub) [inline pages/join/01-you.tsx]: need to add a conditional here based on feedback from @taylorkmho, we should only allow LinkedIn emails if the email is verified

Comment (eparubrub) [reply pages/join/01-you.tsx]: added conditional for Google but not for LinkedIn. Unfortunately the API call for `/v2/userinfo` in LinkedIn is the endpoint needed to indicate whether the user has verified their email, but with the current permissions we have for the LinkedIn API, we're unable to access that endpoint

However, I was able to confirm (via testing it out) that LinkedIn accounts cannot be created without the user verifying their email, so the assumption can be made that LinkedIn accounts have a verified email

File: pages/join/04-contact.tsx [hunk 2]
```diff
@@ -47,7 +52,13 @@ export default function JoinStep4({ pageTitle }) {
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState<ErrorMessageProps>(undefined);
 
+  const [linkedInPicture, setLinkedInPicture] = useState<string>("");
+  const [savePicture, setSavePicture] = useState<boolean>(true);
+  const [user, authLoading, authStateError] = useAuthState(getAuth());
+  const [loginType, setLoginType] = useState<LoginTypeNameEnum>();
+
   const createMember = async () => {
+    const token = await user.getIdToken();
```
Comment (tellaho) [inline pages/join/04-contact.tsx:61]: Ah is this how we do any admin checks?

Comment (eparubrub) [reply pages/join/04-contact.tsx:61]: this isn't necessarily only for admin checks, these JWT tokens are used to verify auth in general

as we talked about on discord, we'll use [custom claims](https://firebase.google.com/docs/auth/admin/custom-claims) to assign admin permissions per user

**Comment by eparubrub** [comment]
Closing this PR as after review with @taylorkmho, users will likely be more inclined to a simpler email auth path. Additionally, it will reduce complexity in authentication and the current Google/LinkedIn authentication methods do not provide much benefit for their implementation and maintenance

This closed PR will serve as a good guide for how to use auth in the future with those integrations if they do become useful

**Pull request closed without merging**